app/controllers/MemberDetailsController.scala

```diff
@@ -69,61 +72,62 @@ trait MemberDetailsController extends RasController {
       },
       memberDetails => {
         Logger.debug("[MemberDetailsController][post] valid form")
-        customerMatchingAPIConnector.findMemberDetails(memberDetails).flatMap { uuid =>
-
-          if (!uuid.isDefined) {
-            Logger.info("[MemberDetailsController][post] UUID not contained in the Location header")
-            Future.successful(Redirect(routes.GlobalErrorController.get))
-          }
 
-          residencyStatusAPIConnector.getResidencyStatus(uuid.get).flatMap { rasResponse =>
-
-            val name = memberDetails.firstName + " " + memberDetails.lastName
-            val dateOfBirth = memberDetails.dateOfBirth.asLocalDate.toString("d MMMM yyyy")
-            val cyResidencyStatus = extractResidencyStatus(rasResponse.currentYearResidencyStatus)
-            val nyResidencyStatus = extractResidencyStatus(rasResponse.nextYearForecastResidencyStatus)
-
-            if (cyResidencyStatus.isEmpty || nyResidencyStatus.isEmpty) {
-              Logger.info("[MemberDetailsController][post] An unknown residency status was returned")
-              Future.successful(Redirect(routes.GlobalErrorController.get))
-            }
-            else {
-
-              Logger.info("[MemberDetailsController][post] Match found")
-
-              val residencyStatusResult = ResidencyStatusResult(cyResidencyStatus, nyResidencyStatus,
-                TaxYearResolver.currentTaxYear.toString, (TaxYearResolver.currentTaxYear + 1).toString,
-                name, dateOfBirth, memberDetails.nino)
-
-              Future.successful(Ok(views.html.match_found(residencyStatusResult)))
+        sessionService.cacheMemberDetails(memberDetails) flatMap {
+          case Some(session) => {
+
+            customerMatchingAPIConnector.findMemberDetails(memberDetails).flatMap { uuid =>
+
+              if (!uuid.isDefined) {
+                Logger.info("[MemberDetailsController][post] UUID not contained in the Location header")
+                Future.successful(Redirect(routes.GlobalErrorController.get))
+              }
+
+              residencyStatusAPIConnector.getResidencyStatus(uuid.get).map { rasResponse =>
+
+                val name = memberDetails.firstName + " " + memberDetails.lastName
+                val dateOfBirth = memberDetails.dateOfBirth.asLocalDate.toString("d MMMM yyyy")
+                val cyResidencyStatus = extractResidencyStatus(rasResponse.currentYearResidencyStatus)
+                val nyResidencyStatus = extractResidencyStatus(rasResponse.nextYearForecastResidencyStatus)
+
+                if (cyResidencyStatus.isEmpty || nyResidencyStatus.isEmpty) {
```

Comment:
Is this what we want to do?

If we get current year should we just return current year

Reply:
Will do this as part of RAS-474

Reply:
Leave for now, more a query to discuss.